src/shared/hooks/use-basic-logging.ts

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useRef } from "react";
+import { log } from "@concord-consortium/lara-interactive-api";
+
+interface IConfig {
+  disabled?: boolean;
+}
+
+export const useBasicLogging = (options?: IConfig) => {
+  const disabled = options?.disabled;
+  const answerToLog = useRef<string | undefined>(undefined);
+
+  useEffect(() => {
+    if (disabled) {
+      return;
+    }
+
+    const focusIn = (e: FocusEvent) => {
+      // event.relatedTarget === null means that user is most likely coming from outer window, and not changing focus
+      // within the interactive iframe (eg clicking various buttons). We want to log only "main" focus event.
+      if (!e.relatedTarget) {
+        log("focus in", { focus_target: (e.target as HTMLElement).tagName.toLowerCase() });
+      }
+    };
+    const focusOut = (e: FocusEvent) => {
+      // event.relatedTarget === null means that user is most likely leaving to outer window, and not changing focus
+      // within the interactive iframe (eg clicking various buttons). We want to log only "main" blur event.
+      if (!e.relatedTarget) {
+        log("focus out");
+        if (answerToLog.current !== undefined) {
+          // LARA uses "answer saved" message for its basic questions too.
+          log("answer saved", { answer_text: answerToLog.current });
```

Comment:
For a multiple choice will the researcher be able to tell the id of the choice made by the student?

If the user closes the browser window is this answer saved event logged? 
I'd think that it might work sometimes but not others, I'd think closing the page might trigger a focus out, but the browser might kill off the code before it can send any messages.

Reply:
> For a multiple choice will the researcher be able to tell the id of the choice made by the student?

No, just the choice text + (correct) (if choice is marked as correct). Will that be a problem? Choice IDs can be long and not very readable (default 3 choices have simple, predefined IDs, but when you add a new one, I think UUID string is used). If we want to have a choice ID as a fallback, we could pass an interactive state too (but that would increase amount of data we send to log system).

> If the user closes the browser window is this answer saved event logged?

I tested that by replacing LOGGER_URI with my local server. It won't save it in practice. It's limitation of saving on blur, but that was a requirement. To make it more reliable, we would probably need to use the same approach that LARA is using to prevent data loss. So, I guess we'd need to make LARA log "answer saved" events instead of interactives. But then we would have more events in the stream and they won't be related to blur/focusout events. In this case, we could add answer_text to "focus out" event here, so researcher would have a choice which kind of events to use.